### DISCUSSION TITLE
PIXI.BLEND_MODES    Can Graphic add its own shader

### DISCUSSION THREAD

**Discussion by pcbren1** [pixijs/pixijs#7590] [Q&A]
![image](https://user-images.githubusercontent.com/86771584/125254894-7cb1e980-e32d-11eb-9505-3a5e534d1b8f.png)

![1234](https://user-images.githubusercontent.com/86771584/125256580-26de4100-e32f-11eb-958f-56ee1d696886.png)

**Comment by ivanpopelyshev** [comment]
You have to use https://github.com/pixijs/pixi-picture/

https://github.com/pixijs/pixi-picture/blob/master/src/ShaderParts.ts

first, make an NPM blend code, like overlay part: https://github.com/pixijs/pixi-picture/blob/aed834cbf877148ad4f9d485f0d23b16b2708784/src/ShaderParts.ts#L20-L40

then make whole fragment: https://github.com/pixijs/pixi-picture/blob/aed834cbf877148ad4f9d485f0d23b16b2708784/src/ShaderParts.ts#L125

then put it in blend array: https://github.com/pixijs/pixi-picture/blob/master/src/ShaderParts.ts#L131

then use a filter.

```js
import {blendFullArray, NPM_BLEND, getBlendFilter} from '@pixi/picture';

const MY_MAGICAL_BLEND = 333;
//...

blendFullArray[MY_MAGICAL_BLEND] = NPM_BLEND.replace(`%NPM_BLEND%`, MY_MAGICAL_SHADER_PART);

const myFilter = getBlendFilter(MY_MAGICAL_BLEND);
```

In fact, I think, I already told you those things in another ticket ;)
